Public/Src/Engine/UnitTests/Scheduler/PipExecutorTest.cs

```diff
@@ -980,6 +980,15 @@ public async Task FailingProcessWithErrorRegex(OutputReportingMode outputReporti
                             "Non-truncated error message was not found in error event. Full output:" + log);
                     }
 
+                    if (outputReportingMode == OutputReportingMode.FullOutputAlways)

```

Comment:
>outputReportingMode == OutputReportingMode.FullOutputAlways [](start = 24, length = 59)

I'm not sure this is the right condition. There are some others that may come into play like FullOutputOnError.

I didn't notice this in the previous iteration of the PR but I think it would be better to make the condition whether the message was truncated rather than the configuration that may impact whether message is truncated. For example even if you have truncation enabled (the default) the error might only be 1 line long. Then you wouldn't have a need to open up the .err file since all of its contents would already be in the error message.